v2/workloadapi/client.go

```diff
@@ -2,75 +2,299 @@ package workloadapi
 
 import (
 	"context"
+	"errors"
 
+	"github.com/spiffe/go-spiffe/proto/spiffe/workload"
 	"github.com/spiffe/go-spiffe/v2/bundle/jwtbundle"
 	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
+	"github.com/spiffe/go-spiffe/v2/logger"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
 	"github.com/spiffe/go-spiffe/v2/svid/jwtsvid"
 	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 )
 
 // Client is a Workload API client.
-type Client struct{}
+type Client struct {
+	connection *grpc.ClientConn
+	wlClient   workload.SpiffeWorkloadAPIClient
+	copts      clientOption
+}
 
 // New dials the Workload API and returns a client.
 func New(ctx context.Context, options ...ClientOption) (*Client, error) {
-	panic("not implemented")
+	c := &Client{
+		copts: defaultClientOptions(),
+	}
+	for _, opt := range options {
+		opt.apply(&c.copts)
+	}
+
+	if c.copts.address == "" {
+		var ok bool
+		c.copts.address, ok = GetDefaultAddress()
+		if !ok {
+			return nil, errors.New("workload endpoint socket address is not configured")
+		}
+	}
+
+	var err error
+	c.copts.address, err = parseTargetFromAddr(c.copts.address)
+	if err != nil {
+		return nil, err
+	}
+
+	c.copts.dialOptions = append(c.copts.dialOptions, grpc.WithInsecure())
+	c.connection, err = grpc.DialContext(ctx, c.copts.address, c.copts.dialOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	c.wlClient = workload.NewSpiffeWorkloadAPIClient(c.connection)
+	return c, nil
 }
 
 // Close closes the client.
 func (c *Client) Close() error {
-	panic("not implemented")
+	return c.connection.Close()
 }
 
 // FetchX509SVID fetches the default X509-SVID, i.e. the first in the list
 // returned by the Workload API.
 func (c *Client) FetchX509SVID(ctx context.Context) (*x509svid.SVID, error) {
-	panic("not implemented")
+	grpcCtx := newGrpcContext(ctx)
+	stream, err := c.wlClient.FetchX509SVID(grpcCtx, &workload.X509SVIDRequest{})
+	if err != nil {
+		return nil, err
+	}
+	resp, err := stream.Recv()
+	if err != nil {
+		return nil, err
+	}
+	svid := resp.Svids[0]
```

Comment:
We should be more defensive here. Although the workload API should never return an empty `svid` field, this should validate the input and fail gracefully instead of panicking.